R/transformCounts.R

```diff
@@ -380,14 +378,15 @@ setMethod("relAbundanceCounts",signature = c(x = "SummarizedExperiment"),
         stop("'pseudocount' must be FALSE or a single numeric value.",
              call. = FALSE)
     } else if(is.numeric(pseudocount)){
-        if (method == "relabundance" && pseudocount > 0){
-            warning("Relative abundances vary in [0, 1]; adding a ",
-                    "pseudocount > 0 on relabundance will cause ",
+        if(  all(mat >= 0) && all(mat <= 1) && pseudocount > 0 ){
+            warning("The abundance table vary in [0, 1]; adding a ",
+                    "pseudocount > 1 on relabundance will cause ",
                     "non-sensicale results. Recommended to cross-check ",
                     "that the pseudocount choice is correct and intended.",
                     call. = FALSE)
         }
```

Comment:
What is the purpose if this?

Pseudocount > 0, should not be added when doing relative transformation? (But negative is fine?).

Now I modified it so that when we have relative abundances, pseudocount > 0 should not be added (so that colSum 1 is not exceed)

However, I started to think what is the purpose of this, because at least in the case of CLR, we need to add pseudocount, and colSum will not be 1.

Reply:
I think this was intended to avoid situation where already relative data is transformed so that colSums!=1. 

But for CLR this is not a problem, colSums can be anything.

There could be a warning that pseudocount is changing the relative ratios but I tend to think that this is not needed as it is sort of self-evident that pseudocount is leading to changes in the data in all possible scenarios.

I would just remove this warning and enable pseudocount.

Reply:
Additonally: I changed default value from pseudocount = FALSE to pseudocount = NULL

I think it is better because the value can be "numeric" or "no". (Not "numeric", "no", and "yes")